### PR TITLE
added homework 10-final

### PR DESCRIPTION
**Pull request by DavidTereshchenko**
https://davidtereshchenko.github.io/Game-Project/

### CODE REVIEW

**Comment by anna-vrana-moc** [comment]
Hi @DavidTereshchenko please provide the link to you project in the description here

**Comment by DavidTereshchenko** [comment]
> Hi @DavidTereshchenko please provide the link to you project in the description here

https://davidtereshchenko.github.io/Game-Project/

**Comment by anna-vrana-moc** [comment]
Happy Hew Year @DavidTereshchenko!
I'm starting the review
1) You can improve responsive like
Screenshot 2024-01-01 at 05 56 53

File: docs/DavidTereshchenko/calculator.html [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                        <span></span>
+                    </div>
+                </div>
+            </header>
+            <main class="page">
+                <div class="main__container">
+                    <div class="title__calculator">
+                        <h1>Calculate Games</h1>
+                    </div>
+                    <div class="calculator__container">
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/calculator.html:48]: It is an unnecessary wrap. Please remove it

File: docs/DavidTereshchenko/calculator.html [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                        <div class="number__container">
+                            <div class="number-1">
+                                <label for="firstNumber">First number</label>
+                                <input id="firstNumber" data-number-one="one" class="number__block one" name="number1" required>
+                            </div>
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/calculator.html:53]: You can remove  `<div class="number-1">` and simplify your code:
```
<label for="firstNumber">First number
       <input id="firstNumber" data-number-one="one" class="number__block one" name="number1" required>
</label>
```

File: docs/DavidTereshchenko/calculator.html [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                            <div class="operator__block">
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/calculator.html:54]: It is an unnecessary wrap. Please remove it

File: docs/DavidTereshchenko/calculator.html [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                                        <option value="/">/</option>
+                                        <option value="%">%</option>
+                                        <option value="**">**</option>
+                                    </select>
+                                </label>
+                            </div>
+                            <div class="number-2">
+                                <label for="secondNumber">Second number</label>
+                                <input id="secondNumber" data-number-two="two" class="number__block two" name="number2" required>
+                            </div>
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/calculator.html:69]: The same story:
You can remove `<div class="number-2">` and simplify your code:

```
<label for="secondNumber">First number
      <input id="secondNumber" data-number-two ="two" class="number__block one" name="number2" required>
</label>
```

File: docs/DavidTereshchenko/calculator.html [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+                            <div class="return">
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/calculator.html:70]: It is an unnecessary wrap. Please remove it

File: docs/DavidTereshchenko/calculator.html [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                                <button type="button" class="button" data-button="button-result">=</button>
+                            </div>
+                            <div class="result__block">
+                                <p class="resul__number result"></p>
+                                <span class="count-games" data-count-game></span>
+                            </div>
+                        </div>
+                        <div class="count-block" data-count-block="count-block">
+                            <div class="visual-counting">
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/calculator.html:79]: The div above is an unnecessary wrap. Please remove it and add `class="count-block" data-count-block="count-block"` to `visual-counting` block

File: docs/DavidTereshchenko/css/calculator.css [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+  justify-content: center;
+}
+
+.visual-counting span {
+  font-size: 48px;
+  color: #EF4934;
+}
+
+.date {
+  display: flex;
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/calculator.css:124]: This property is useful for aligning blocks. Avoid using it for strings

File: docs/DavidTereshchenko/css/calculator.css [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+  border: 1px solid #EF4934;
+}
+
+.oper {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.visual-counting span {
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/calculator.css:118]: You'll get the same result without the span. Simplify the code and avoid specifying unnecessary styles

File: docs/DavidTereshchenko/css/calculator.css [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+  flex-direction: row;
+  gap: 10px;
+}
+
+.result {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding-bottom: 5px;
+}
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/calculator.css:84]: You can remove 
  ```
display: flex;
  flex-direction: row;
  align-items: center;
```
 and keep the same visual. So let's do it)

File: docs/DavidTereshchenko/css/calculator.css [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+.result p {
+  display: flex;
+  align-items: center;
+  font-size: 14px;
+}
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/calculator.css:90]: You need to specify only `font-size: 14px; ` here, so you can remove unnecessary code

File: docs/DavidTereshchenko/css/calculator.css [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+
+.count-block {
+  display: none;
+  flex-direction: row;
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/calculator.css:94]: by default `display: flex` has `flex-direction: row; ` so you don't need to write it unless you switch between column and row layouts

File: docs/DavidTereshchenko/css/calculator.css [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+  margin-top: 40px;
+  gap: 20px;
+  align-items: center;
+}
+
+.visual-counting {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 20px;
+  flex-direction: row;
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/calculator.css:104]: please remove ` flex-direction: row;`

File: docs/DavidTereshchenko/css/calculator.css [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+
+.error {
+  border: 1px solid #EF4934;
+}
+
+.oper {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/calculator.css:116]: I believe, you can remove all styles:
```
.oper {
  display: flex;
  align-items: center;
  justify-content: center;
}
```

File: docs/DavidTereshchenko/calculator.html [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+                                <span class="number__1" data-number-one="number-one"></span>
+                                <span class="oper" data-operation-visual="operation-visual"></span>
+                                <span class="number__2" data-number-two="number-two"></span>
+                                <span class="equals" data-equals="equals"></span>
+                                <span class="visual-result" data-result-visual="visual-result"></span>
+                                <span class="count-games__visual" data-count-game-visual="count-game"></span>
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/calculator.html:85]: You can remove classes from each `span` since you don't have any specific styles for them

**Review by anna-vrana-moc — CHANGES_REQUESTED** [review]

**Comment by anna-vrana-moc** [comment]
I like your mobile filter icon
Screenshot 2024-01-01 at 06 52 57

**Comment by anna-vrana-moc** [comment]
2) Please fix the default outline on the select:
Screenshot 2024-01-01 at 06 55 46

File: docs/DavidTereshchenko/script/cards.js
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/script/cards.js:1]: It is better to use similar names for all files on the page. So, rename this file to `'game.js'` like `'game.html'`, `'game.css'`

File: docs/DavidTereshchenko/script/cards.js [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+    setTimeout(() => {
+        hideLoader();
+        createCard(filterCards);
+    }, 1000);
+}
+
+checkBoxFilter(dateYear);
+platformFilter();
+categoryFilter();
+sortFilter();
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/script/cards.js:252]: You can use `'function  init'` to call all functions

File: docs/DavidTereshchenko/script/cards.js [hunk 16]
```diff
@@ -0,0 +1,252 @@
...
+    cardsBlock.innerText = '';
+
+    setTimeout(() => {
+        hideLoader();
+        createCard(games);
+    }, 1000);
+}
+
+document.querySelector('[data-search]').addEventListener('click', () => {
+    showLoader();
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/script/cards.js:186]: You can create the `function search` and just use it here

**Comment by anna-vrana-moc** [comment]
3) Please add common padding of 15px here also
Screenshot 2024-01-01 at 07 21 59

File: docs/DavidTereshchenko/css/successful.css [hunk 17]
```diff
@@ -0,0 +1,232 @@
...
+  justify-content: center;
+  align-items: center;
+}
+
+.success {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  row-gap: 30px;
+}
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/css/successful.css:32]: You can remove this styles

File: docs/DavidTereshchenko/successful.html [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+                    <div class="success-block">
+                        <div class="success-checkmark">
+                            <div class="check-icon">
+                                <span class="icon-line line-tip"></span>
+                                <span class="icon-line line-long"></span>
+                                <div class="icon-circle"></div>
+                                <div class="icon-fix"></div>
+                            </div>
+                        </div>
+                        <div class="success">
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/successful.html:51]: You can remove this wrap and just use text-align: center for `success-block`

File: docs/DavidTereshchenko/successful.html [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+                            <h1 class="success__title">Successful</h1>
+                            <p class="success__info">You can continue watching your favorite games</p>
+                        </div>
+                        <form action="game.html" class="success__form">
```
Comment (anna-vrana-moc) [inline docs/DavidTereshchenko/successful.html:55]: You can add the class `success__back-home` and remove the block below

**Review by anna-vrana-moc — CHANGES_REQUESTED** [review]

**Review by anna-vrana-moc — DISMISSED** [review]

**Comment by anna-vrana-moc** [comment]
@DavidTereshchenko thank you for the all fixes. You are ready for final review! @MZabolotnev if you have a chance, please take a look and add your review :)

File: docs/SerhiiNSN/calculator.html
Comment (vladyslav-yermolin-moc) [inline docs/SerhiiNSN/calculator.html:1]: Hey, @DavidTereshchenko, I could merge your pr, but you deleted the work of your colleague. Please move these files in the folder `SerhiiNSN` back and I' merge your pr
image

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]

File: docs/SerhiiNSN/calculator.html
Comment (anna-vrana-moc) [reply docs/SerhiiNSN/calculator.html:1]: it is fixed now

**Review by anna-vrana-moc — APPROVED** [review]

**Review by vladyslav-yermolin-moc — APPROVED** [review]

**Pull request merged**